iModelCore/ECPresentation/Source/Content/ContentQueryResultsReader.cpp

```diff
@@ -416,17 +416,18 @@ void DistinctValuesAccumulator::ReadPrimitivePropertyRecord(ContentDescriptor::E
         ECValue rawValue = ValueHelpers::GetECValueFromSqlValue(prop.GetProperty().GetAsPrimitiveProperty()->GetType(), extendedType, value);
 
         auto it = currentRecords.find(displayValue);
+        auto entry = std::make_pair(rawValue, &extendedType);
         if (it == currentRecords.end())
-            currentRecords.emplace(displayValue, bvector<std::pair<ECValue, Utf8String>>({std::make_pair(rawValue, extendedType)}));
-        else if (!ContainerHelpers::Contains(it->second, std::make_pair(rawValue, extendedType)))
-            it->second.push_back(std::make_pair(rawValue, extendedType));
+            currentRecords.emplace(displayValue, bvector<std::pair<ECValue, Utf8StringCP>>({entry}));
+        else if (!ContainerHelpers::Contains(it->second, [entry](auto record) {return entry.first.Equals(record.first) && entry.second->Equals(record.second->c_str()); }))

```

Comment:
```suggestion
        else if (!ContainerHelpers::Contains(it->second, [&entry](auto const& record) {return entry.first.Equals(record.first) && entry.second->Equals(record.second->c_str());}))
```

avoid unnecessary copies

Reply:
fixed